File: csharp/src/ReservationRequestDto.cs
namespace TrainKata
{
    public class ReservationRequestDto
    {
        public string TrainId { get; }
        public int SeatCount { get; }

        public ReservationRequestDto(string trainId, int seatCount)
        {
            TrainId = trainId;
            SeatCount = seatCount;
        }

        public ReservationRequest ToRequest()
        {
            return new ReservationRequest(new TrainId(TrainId), SeatCount);
        }
    }
}

File: Java/src/main/java/ui/ReservationRequestDto.java
package ui;

import domain.ReservationRequest;
import domain.SeatCount;
import domain.TrainId;

public class ReservationRequestDto {
	public final String trainId;
    public final int seatCount;

    public ReservationRequestDto(String trainId, int seatCount) {
		this.trainId = trainId;
        this.seatCount = seatCount;
    }

    public ReservationRequest toDomainModel(){
       return new ReservationRequest(new TrainId(trainId), new SeatCount(seatCount));
    }
}